lib/buffer.js

```diff
@@ -339,10 +339,14 @@ function slowToString(encoding, start, end) {
 
 
 Buffer.prototype.toString = function() {
-  const length = this.length | 0;
-  if (arguments.length === 0)
-    return this.utf8Slice(0, length);
-  return slowToString.apply(this, arguments);
+  if (arguments.length === 0) {
+    var result = this.utf8Slice(0, this.length);
```

Comment:
Good catch. Though not going to let it hold up the PR. Whoever merges it can fix it, if it hasn't been fixed before being merged.